prow/cmd/job-forker/forking.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"errors"
+	"path"
+
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	ghi "github.com/gardener/ci-infra/prow/pkg/githubinteractor"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+
+	"sigs.k8s.io/yaml"
+)
+
+func generateVersionsFromBranches(branches []github.Branch, branchPrefix string) []string {
+	versions := make([]string, len(branches))
+	for i, branch := range branches {
+		versions[i] = strings.ReplaceAll(branch.Name, branchPrefix, "")
+	}
+
+	return versions
+}
+
+func generatePresubmits(j config.JobConfig, version, repository string) []config.Presubmit {
+	newPresubmits := []config.Presubmit{}
+	for repo, presubmits := range j.PresubmitsStatic {
+		if repo != repository {
+			continue
+		}
+		for _, presubmit := range presubmits {
+			if presubmit.Annotations[forkAnnotation] != "true" {
```

Comment:
Is there any reason not to get the array directly by key from the map?

```suggestion
	for _, presubmit := range j.PresubmitsStatic[repository] {
			if presubmit.Annotations[forkAnnotation] != "true" {
```